### PR TITLE
ITKDev - patron menu v3

### PR DESCRIPTION
**Pull request by jeppekroghitk**
#### Link to issue

N/A

#### Description

Add patron menu overlay and hook into the top-bar user icon.

#### Screenshot of the result

N/A

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

N/A

### CODE REVIEW

**Comment by cableman** [comment]
Re-based after merge of reservation list PR.

File: web/modules/custom/dpl_patron_menu/src/Plugin/Block/PatronMenuBlock.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      ],
+    ];
+
+    $data = [
+      // Config.
+      "threshold-config" => $this->configFactory->get('dpl_library_agency.general_settings')->get('threshold_config'),
+      "menu-navigation-data-config" => json_encode($menu, JSON_THROW_ON_ERROR),
+      // Added in THEME_preprocess_dpl_react_app__menu to use the theme defined
+      // icon.
+      "profile_svg" => '',
```
Comment (spaceo) [inline web/modules/custom/dpl_patron_menu/src/Plugin/Block/PatronMenuBlock.php:115]: This is missing, right?

File: web/themes/custom/novel/novel.theme [hunk 2]
```diff
@@ -83,3 +83,23 @@ function novel_preprocess_dpl_react_app__search_result(array &$variables): void
...
+ * @throws \Safe\Exceptions\StringsException
+ */
+function novel_preprocess_dpl_react_app__menu(array &$variables): void {
+  $profile_svg = file_get_contents(sprintf('%s/%s',
+    \Drupal::theme()->getActiveTheme()->getPath(),
+    '/assets/dpl-design-system/icons/collection/Profile.svg'
+  ));
+
+  $variables += [
+    'profile_svg' => $profile_svg ?? NULL,
```
Comment (spaceo) [inline web/themes/custom/novel/novel.theme:103]: Where is the theme variable being used?

File: web/modules/custom/dpl_patron_menu/dpl_patron_menu.info.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+description: Handle the patron user menu
+package: DPL
+dependencies:
+  - drupal:block
+  - dpl_library_agency:dpl_library_agency
+  - dpl_loans:dpl_loans
+#  - dpl_reservation:dpl_reservation
+#  - dpl_fees:dpl_fees
+#  - dpl_favorites_list:dpl_favorites_list
+#  - dpl_dashboard:dpl_dashboard
```
Comment (spaceo) [inline web/modules/custom/dpl_patron_menu/dpl_patron_menu.info.yml:11]: If they are not used I think they should be removed. Do you agree?

File: web/modules/custom/dpl_patron_menu/dpl_patron_menu.module [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+function dpl_patron_menu_preprocess_page(array &$variables): void {
+  $blockManager = \Drupal::getContainer()->get('plugin.manager.block');
+
+  /** @var \Drupal\dpl_patron_menu\Plugin\Block\PatronMenuBlock $plugin_block */
+  $plugin_block = $blockManager->createInstance('dpl_patron_menu_block', []);
+
+  $access_result = $plugin_block->access(\Drupal::currentUser());
+  if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+    throw new AccessDeniedHttpException();
+  }
```
Comment (spaceo) [inline web/modules/custom/dpl_patron_menu/dpl_patron_menu.module:27]: Please add a todo on you know what

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks good. I have some questions though.

File: web/themes/custom/novel/novel.theme (hunk 2 above)
Comment (cableman) [reply web/themes/custom/novel/novel.theme:103]: It's used when rendering the block. But changed able by the theme. Trying to use the same pattern as you used for `novel_preprocess_dpl_react_app__search_header`

File: web/modules/custom/dpl_patron_menu/src/Plugin/Block/PatronMenuBlock.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/dpl_patron_menu/src/Plugin/Block/PatronMenuBlock.php]: yeah, something has fallen between backend/frontend... we will fix it

File: web/themes/custom/novel/novel.theme (hunk 2 above)
Comment (cableman) [reply web/themes/custom/novel/novel.theme]: Removed

File: web/modules/custom/dpl_patron_menu/dpl_patron_menu.module (hunk 4 above)
Comment (cableman) [reply web/modules/custom/dpl_patron_menu/dpl_patron_menu.module:28]: Added todo

File: web/modules/custom/dpl_patron_menu/dpl_patron_menu.info.yml (hunk 3 above)
Comment (cableman) [reply web/modules/custom/dpl_patron_menu/dpl_patron_menu.info.yml]: Added the dependencies as they are required to render menu links based on routes from these modules.

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**